x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/analysis/PostAnalyzer.java

```diff
@@ -49,24 +47,21 @@ public LogicalPlan postAnalyze(LogicalPlan plan, EqlConfiguration configuration)
             Holder<Boolean> hasJoin = new Holder<>(Boolean.FALSE);
 
             Source projectCtx = synthetic("<implicit-project>");
-            if (plan.anyMatch(Sequence.class::isInstance)) {
+            // first per KeyedFilter
+            plan = plan.transformUp(KeyedFilter.class, k -> {
                 hasJoin.set(Boolean.TRUE);
-                // first per KeyedFilter
-                plan = plan.transformUp(KeyedFilter.class, k -> {
-                    Project p = new Project(projectCtx, k.child(), k.extractionAttributes());
+                Project p = new Project(projectCtx, k.child(), k.extractionAttributes());
 
-                    // TODO: this could be incorporated into the query generation
-                    LogicalPlan fetchSize = new LimitWithOffset(
-                        synthetic("<fetch-size>"),
-                        new Literal(synthetic("<fetch-value>"), configuration.fetchSize(), DataTypes.INTEGER),
-                        p
-                    );
+                // TODO: this could be incorporated into the query generation
+                LogicalPlan fetchSize = new LimitWithOffset(
```

Comment:
Also, this is not needed for samples. The `fetch_size` is passed directly (through the EqlConfiguration instance) in the ExecutionManager. Maybe use a simple projection for samples:

```
                plan = plan.transformUp(KeyedFilter.class, k -> {
                    Project p = new Project(projectCtx, k.child(), k.keys());
                    return new KeyedFilter(k.source(), p, k.keys(), k.timestamp(), k.tiebreaker());
                });
```